polaris-react/src/components/ActionList/ActionList.tsx

```diff
@@ -132,11 +130,17 @@ export function ActionList({
     return totalSectionItems;
   }, [filteredSections]);
 
-  const showSearch = totalActions >= 8;
+  const totalActions =
+    finalSections?.reduce(
+      (acc: number, section) => acc + section.items.length,
+      0,
+    ) || 0;
+
+  const hasSearch = totalActions >= 8;
```

Comment:
This variable name doesn't make as much sense as it used to. Can we rename it and also move `8` to a const up near the top of the file? Just above the function decl

```suggestion
  const hasManyActions = totalActions >= FILTER_ACTIONS_THRESHOLD;
```